### PR TITLE
optimization: 优化CPU架构获取和检查 (close #530)

### LINKED ISSUE #530: [FEATURE] 升级脚本检查CPU架构
**你想要什么功能**

**为什么需要**

**建议如何实现**

### PR DESCRIPTION
**Pull request by zhangzhw8**

### CODE REVIEW

File: script_tools/upgrade_agent.sh.tpl [hunk 1]
```diff
@@ -3,6 +3,24 @@ OS_INFO=""
...
+        CPU_ARCH="x86"
+        return 0
+    elif [[ "$CPU_ARCH" =~ "aarch" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}}
+
+get_cpu_arch "uname -p" || get_cpu_arch "uname -m"  || arch || echo get_cpu_arch "Failed to get CPU arch, please contact the developer."
```
Comment (ladymamawang) [inline script_tools/upgrade_agent.sh.tpl:22]: get_cpu_arch "uname -p" || get_cpu_arch "uname -m" || arch || fail get_cpu_arch FAILED "Failed to get CPU arch, please contact the developer."

Comment (CohleRustW) [inline script_tools/upgrade_agent.sh.tpl]: get_cpu_arch 这个可以删掉

File: script_tools/upgrade_agent.sh.tpl [hunk 2]
```diff
@@ -94,6 +112,12 @@ remove_crontab () {{
     fi
 }}
 
+if [[ {package_name} == *$CPU_ARCH* ]];then
+        echo "pkg:{package_name} and arch:$CPU_ARCH is ok"
+else
+        echo "pkg:{package_name} and arch:$CPU_ARCH is bad"
+        exit 1
+fi
```
Comment (CohleRustW) [inline script_tools/upgrade_agent.sh.tpl:120]: if grep "${CPU_ARCH}" <<<{package_name} > /dev/null; then
        echo "pkg:{package_name} and arch:$CPU_ARCH is ok"
else
        echo "pkg:{package_name} and arch:$CPU_ARCH is bad"
        exit 1
fi
这样更直观些

File: script_tools/upgrade_agent.sh.tpl (hunk 1 above)
Comment (CohleRustW) [reply script_tools/upgrade_agent.sh.tpl]: get_cpu_arch "uname -p" || get_cpu_arch "uname -m"  || arch || echo "Failed to get CPU arch, please contact the developer."

Comment (zhangzhw8) [reply script_tools/upgrade_agent.sh.tpl]: 中间还漏了，应该这样
get_cpu_arch "uname -p" || get_cpu_arch "uname -m" || get_cpu_arch "arch" || echo "Failed to get CPU arch, please contact the developer."

**Review by CohleRustW — APPROVED** [review]

**Pull request merged**